src/SDKs/ContainerInstance/ContainerInstance.Tests/ContainerGroupScenarioTests.cs

```diff
@@ -0,0 +1,107 @@
+				{
+					Location = resourceGroup.Location,
+					OsType = containerOsType,
+					Containers = new List<Container>
+					{
+						new Container
+						{
+							Name = containerName,
+							Image = containerImage,
+							Ports = new List<ContainerPort>
+							{
+								new ContainerPort(containerPort)
+							},
+							Resources = new ResourceRequirements
+							{
+								Requests = new ResourceRequests
+								{
+									Cpu = containerCpu,
+									MemoryInGB = containerMemoryInGB
+								}
+							}
+						}
+					}
+				});
+				Assert.NotNull(createdContainerGroup);
+				Assert.Equal(containerGroupName, createdContainerGroup.Name);
+				Assert.Equal(containerOsType, createdContainerGroup.OsType);
+				Assert.Equal(1, createdContainerGroup.Containers.Count);
+
+				Container createdContainer = createdContainerGroup.Containers.First();
+				Assert.Equal(containerName, createdContainer.Name);
+				Assert.Equal(containerImage, createdContainer.Image);
+				Assert.Equal(1, createdContainer.Ports.Count);
+				Assert.Equal(containerPort, createdContainer.Ports.First().Port);
+				Assert.Equal(containerCpu, createdContainer.Resources.Requests.Cpu);
+				Assert.Equal(containerMemoryInGB, createdContainer.Resources.Requests.MemoryInGB);
+
+				// Wait till the container group is ready before proceeding
+				ContainerGroupUtilities.WaitTillProvisioningStateSucceeded(containerClient, resourceGroup.Name, containerGroupName);
+
+				string containerGroupID = createdContainerGroup.Id;
+
+				// Get the container group
+				ContainerGroup getContainerGroup = containerClient.ContainerGroups.Get(resourceGroup.Name, containerGroupName);
+				Assert.NotNull(createdContainerGroup);
+				Assert.Equal(containerGroupName, getContainerGroup.Name);
+				Assert.Equal(containerOsType, getContainerGroup.OsType);
+
+				Container getContainer = getContainerGroup.Containers.First();
+				Assert.Equal(containerName, getContainer.Name);
+				Assert.Equal(containerImage, getContainer.Image);
+				Assert.Equal(1, getContainer.Ports.Count);
+				Assert.Equal(containerPort, getContainer.Ports.First().Port);
+				Assert.Equal(containerCpu, getContainer.Resources.Requests.Cpu);
+				Assert.Equal(containerMemoryInGB, getContainer.Resources.Requests.MemoryInGB);
+
+				// List the container group within the resource group
+				IPage<ContainerGroup> listRgContainerGroup = containerClient.ContainerGroups.ListByResourceGroup(resourceGroup.Name);
+				Assert.NotNull(listRgContainerGroup);
+				Assert.Equal(1, listRgContainerGroup.Count());
+				Assert.Equal(containerGroupName, listRgContainerGroup.First().Name);
+
+				// List the container group within the subscription
+				IPage<ContainerGroup> listSubContainerGroup = containerClient.ContainerGroups.List();
+				Assert.NotNull(listRgContainerGroup);
+				Assert.True(listRgContainerGroup.Count() >= 1);
+				Assert.True(listRgContainerGroup.Any(x => containerGroupID.Equals(x?.Id)));
```

Comment:
Create a test case in which you create 2 container group. When listing all, you should get both. When listing by resource group, you should only get one.